### PR TITLE
Use the charmstore own testing server in tests.

### PR DESCRIPTION
**Pull request by frankban**
This must land after https://github.com/juju/charmstore/pull/330

### CODE REVIEW

**Comment by jrwren** [comment]
:+1:

File: charmrepo/charmstore_test.go [hunk 1]
```diff
@@ -94,18 +94,55 @@ type charmStoreRepoSuite struct {
...
+
+	getDownloads := func() int {
+		var result []params.Statistic
+		err := client.Get(path, &result)
+		c.Assert(err, jc.ErrorIsNil)
+		return int(result[0].Count)
+	}
+
+	for retry := 0; retry < 10; retry++ {
+		time.Sleep(100 * time.Millisecond)
```
Comment (hatched) [inline charmrepo/charmstore_test.go]: Is waiting the best way to do this? Should we not instead wait on a trigger of some sort?

**Comment by hatched** [comment]
Thanks! I have just one small question if there is no better way then :+1:

File: charmrepo/charmstore_test.go (hunk 1 above)
Comment (frankban) [inline charmrepo/charmstore_test.go]: Yes, we are on the client side, and so we don't have events from the server.

**Comment by frankban** [comment]
Thanks for the reviews!

File: charmrepo/charmstore_test.go [hunk 2]
```diff
@@ -77,14 +77,14 @@ func (s *charmStoreBaseSuite) TearDownTest(c *gc.C) {
...
-		User:     serverParams.AuthUsername,
-		Password: serverParams.AuthPassword,
-	})
-	ch := TestCharms.CharmDir(name)
-	id, err := client.UploadCharm(charm.MustParseReference(url), ch)
-	c.Assert(err, jc.ErrorIsNil)
+	id := charm.MustParseReference(url)
+	promulgated := false
+	if id.User == "" {
+		id.User = "who"
```
Comment (urosj) [inline charmrepo/charmstore_test.go]: Why do you set a user to "who"?

**Pull request merged**